### PR TITLE
10 parking raspberrypi

### LINKED ISSUE #10: parking (raspberrypi)
- when sensor detect something (proximity)
- send message parking spot occupied
- turn on red light (closed loop)
- when sensor detect nothing
- send message parking spot free
- turn on green light (closed loop)

use a mqtt queue, 1 topic per sensor (2)

### PR DESCRIPTION
**Pull request by Solargi**
implemented the following feartures:
when sensor detect something (proximity)
send message parking spot occupied
turn on red light (closed loop)
when sensor detect nothing
send message parking spot free
turn on green light (closed loop)

the variable car_detection_distance used to set a distance from which the car is considered detected must be adjusted when assembling the lego city, there is also a sleep(2) which is used just to avoid overloading the public mqtt server, in order to avoid a possible ban. the sleep command can be changed and/or removed when using the private mqtt broker. 
the message parking spot occupied is published as an attribute of the proximity sensor, the name of the attribute is: parking_spot_status and the value are either "free" or "occupied".
the led are controlled by the raspberry entirely. 

a config.ini file has also been defined

### CODE REVIEW

File: parking/config.ini [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+[parking]
+ps_pin= 2
+ps_id= ps
+red_l_pin=3
+red_l_id= red_l
+green_l_pin= 4
+green_l_id = green_l
+parking_service_api_key= test_key
+car_detection_distance=4
```
Comment (chicco785) [inline parking/config.ini]: aggiungi la linea a fine codice

File: parking/parking.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            elif car_dist > car_detection_distance and parking_spot_status == "occupied":
+                # set parking status to free
+                parking_spot_status = "free"
+                # send message to update parking status
+                pub_parking_status(parking_spot_status,
+                                   ps_topic, auth, broker_address, port)
+                # turn on red turn of green
+                set_led(green_l_pin)
+            # used to limit requests to public mqtt broker to avoid getting banned to be removed when using private mqtt broker
+            sleep(2)
```
Comment (chicco785) [inline parking/parking.py]: facciamo configurabile via variabile

**Review by chicco785 — CHANGES_REQUESTED** [review]

File: parking/config.ini (hunk 1 above)
Comment (Solargi) [reply parking/config.ini]: done

File: parking/parking.py (hunk 2 above)
Comment (Solargi) [reply parking/parking.py]: done

**Review by chicco785 — APPROVED** [review]

**Pull request merged**